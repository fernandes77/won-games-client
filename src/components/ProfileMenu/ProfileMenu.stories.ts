import type { Meta, StoryObj } from '@storybook/react'
import { ProfileMenu } from './ProfileMenu'

const meta: Meta<typeof ProfileMenu> = {
  component: ProfileMenu
}

export default meta
type Story = StoryObj<typeof ProfileMenu>

export const Primary: Story = {}
