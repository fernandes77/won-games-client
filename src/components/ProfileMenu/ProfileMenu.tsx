import {
  IconCreditCard,
  IconList,
  IconLogout,
  IconUser
} from '@tabler/icons-react'
import { Url } from 'next/dist/shared/lib/router/router'
import Link from 'next/link'
import { useRouter } from 'next/router'

const ProfileMenuLink = ({
  icon,
  label,
  href
}: { href: Url; icon: React.ReactNode; label: string }) => {
  const router = useRouter()

  const cssActive = 'bg-primary text-white'
  return (
    <Link
      href={href}
      className={`bg-white text-black flex items-center py-xs px-sm
      transition-[background,color,0.3s_ease-in-out] hover:bg-primary hover:text-white
      max-md:justify-center max-md:flex-[1]
      ${router.asPath === href ? cssActive : ''}`}
    >
      {icon}
      <span className="ml-xs max-md:hidden">{label}</span>
    </Link>
  )
}

export const ProfileMenu = () => (
  <nav
    className="flex border-b-[1px] border-gray-light min-md:flex-col min-md:[&_a]:border-b-[1px]
    min-md:[&_a]:border-gray-light last:min-md:[&_a]:border-0"
  >
    <ProfileMenuLink
      href="/profile/me"
      icon={<IconUser size={24} />}
      label="My profile"
    />

    <ProfileMenuLink
      href="/profile/cards"
      icon={<IconCreditCard size={24} />}
      label="My cards"
    />

    <ProfileMenuLink
      href="/profile/orders"
      icon={<IconList size={24} />}
      label="My orders"
    />

    <ProfileMenuLink
      href="/logout"
      icon={<IconLogout size={24} />}
      label="Sign out"
    />
  </nav>
)
