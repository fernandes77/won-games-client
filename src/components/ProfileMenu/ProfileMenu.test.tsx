import { render, screen } from '@/utils/tests/helpers'
import { ProfileMenu } from './ProfileMenu'

describe('<ProfileMenu />', () => {
  it('should render the menu', () => {
    const { container } = render(<ProfileMenu />)

    expect(
      screen.getByRole('link', { name: /my profile/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /my cards/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /my orders/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /sign out/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
