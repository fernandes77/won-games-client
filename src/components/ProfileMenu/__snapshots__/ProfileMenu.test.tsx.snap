// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileMenu /> should render the menu 1`] = `
<nav
  class="flex border-b-[1px] border-gray-light min-md:flex-col min-md:[&_a]:border-b-[1px] min-md:[&_a]:border-gray-light last:min-md:[&_a]:border-0"
>
  <a
    class="bg-white text-black flex items-center py-xs px-sm transition-[background,color,0.3s_ease-in-out] hover:bg-primary hover:text-white max-md:justify-center max-md:flex-[1]"
    href="/profile/me"
  >
    <svg
      class="tabler-icon tabler-icon-user"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
      />
      <path
        d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"
      />
    </svg>
    <span
      class="ml-xs max-md:hidden"
    >
      My profile
    </span>
  </a>
  <a
    class="bg-white text-black flex items-center py-xs px-sm transition-[background,color,0.3s_ease-in-out] hover:bg-primary hover:text-white max-md:justify-center max-md:flex-[1]"
    href="/profile/cards"
  >
    <svg
      class="tabler-icon tabler-icon-credit-card"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 5m0 3a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v8a3 3 0 0 1 -3 3h-12a3 3 0 0 1 -3 -3z"
      />
      <path
        d="M3 10l18 0"
      />
      <path
        d="M7 15l.01 0"
      />
      <path
        d="M11 15l2 0"
      />
    </svg>
    <span
      class="ml-xs max-md:hidden"
    >
      My cards
    </span>
  </a>
  <a
    class="bg-white text-black flex items-center py-xs px-sm transition-[background,color,0.3s_ease-in-out] hover:bg-primary hover:text-white max-md:justify-center max-md:flex-[1]"
    href="/profile/orders"
  >
    <svg
      class="tabler-icon tabler-icon-list"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 6l11 0"
      />
      <path
        d="M9 12l11 0"
      />
      <path
        d="M9 18l11 0"
      />
      <path
        d="M5 6l0 .01"
      />
      <path
        d="M5 12l0 .01"
      />
      <path
        d="M5 18l0 .01"
      />
    </svg>
    <span
      class="ml-xs max-md:hidden"
    >
      My orders
    </span>
  </a>
  <a
    class="bg-white text-black flex items-center py-xs px-sm transition-[background,color,0.3s_ease-in-out] hover:bg-primary hover:text-white max-md:justify-center max-md:flex-[1]"
    href="/logout"
  >
    <svg
      class="tabler-icon tabler-icon-logout"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14 8v-2a2 2 0 0 0 -2 -2h-7a2 2 0 0 0 -2 2v12a2 2 0 0 0 2 2h7a2 2 0 0 0 2 -2v-2"
      />
      <path
        d="M9 12h12l-3 -3"
      />
      <path
        d="M18 15l3 -3"
      />
    </svg>
    <span
      class="ml-xs max-md:hidden"
    >
      Sign out
    </span>
  </a>
</nav>
`;
